src/main/scala/fi/aalto/cs/apluscourses/intellij/Repl.scala

```diff
@@ -22,14 +23,17 @@ class Repl(module: Module) extends ScalaLanguageConsole(module: Module) {
   banner.setVisible(false)
   add(banner, BorderLayout.NORTH)
 
-  // creating a new REPL resets the "module changed" state
-  ReplChangesObserver.onStartedRepl(module)
+  // Do not show the warning banner for non-A+ courses
+  if (PluginSettings.getInstance.getCourseProject(module.getProject) != null) {
```

Comment:
Correct me if I'm wrong: if the REPL is opened for a non-A+ project, and after that, the project is turned into an A+ project, that REPL instance stays "non-tracking". All subsequently opened REPL instances, however, get this functionality. Seems reasonable enough to me.